Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp

```diff
@@ -65,30 +67,32 @@ RemoteRenderingBackendProxy::~RemoteRenderingBackendProxy()
     for (auto& markAsVolatileHandlers : m_markAsVolatileRequests.values())
         markAsVolatileHandlers(false);
 
-    if (!m_gpuProcessConnection)
+    if (!m_streamConnection)
         return;
-    m_gpuProcessConnection->connection().send(Messages::GPUConnectionToWebProcess::ReleaseRenderingBackend(renderingBackendIdentifier()), 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);
+
+    ensureOnMainRunLoop([ident = renderingBackendIdentifier()]() {
```

Comment:
This kind of asynchronousness is a  bit scary.
I can easily imagine a scenario where we start destroying all the objects and shut down the GPU Process. Then this callback gets executed after shutting down the GPUP, and it will create a new GPUP just to send this message.

It would be better to already hold a IPC message and use the thread-safe send of IPC::Connection. However, that's a bit hard if we want to support current auto revival of dead GPUPs